src/core/public/injected_metadata/deep_freeze.ts

```diff
@@ -20,7 +20,7 @@
 type Freezable = { [k: string]: any } | any[];
 
 // if we define this inside RecursiveReadonly TypeScript complains
-interface RecursiveReadonlyArray<T> extends ReadonlyArray<RecursiveReadonly<T>> {}
+interface RecursiveReadonlyArray<T> extends Array<RecursiveReadonly<T>> {}
```

Comment:
@epixa I've been trying to figure out how we can communicate to people consuming the `injectedMetadata` that it is frozen via the types, but while this was working it caused an issue when using `injectedMetadata.getInjectedVar(s)` because `ReadonlyArray<T> !== Array<T>`. At this point I think what was supposed to be helpful is now just getting in the way, so I've switched this to stop using `ReadonlyArray<T>`, happy to do it in a separate PR if you like.

Reply:
Works for me